src/main/java/org/usfirst/frc4904/robot/commands/controlpanel/ColorSensor.java

```diff
@@ -0,0 +1,223 @@
+ */
+public abstract class ColorSensor extends SequentialCommandGroup {
+    public final Color[] colorOrder = new Color[] { Color.RED, Color.YELLOW, Color.BLUE, Color.GREEN };
+
+    public Color currentColor = null;
+    private int colorsPassed = 0;
+    public boolean doneSpinning = false;
+
+    public void trackColorChange() {
+        if (currentColor == null) {
+            currentColor = getColor();
+            return;
+        }
+        Color tempColor = getColor();
+        if (currentColor != tempColor) {
+            Color oldColor = currentColor;
+            currentColor = tempColor;
+            colorsPassed++;
+            if (getIndex(currentColor) == 3) {
+                if (currentColor == colorOrder[0]) {
+                    if (colorsPassed == 23) {
+                        doneSpinning = true;
+                    }
+                } else {
+                    LogKitten.wtf("A color somehow was skipped and we have no code to handle it! :)");
+                    // it is very unlikely for a color to be skipped because the control panel will
+                    // not be rotating that quickly
+                }
+            } else {
+                if (currentColor == colorOrder[getIndex(oldColor) + 1]) {
+                    if (colorsPassed == 23) {
+                        doneSpinning = true;
+                    }
+                } else {
+                    LogKitten.wtf("A color somehow was skipped and we have no code to handle it! :) please consider not dying");
+                    // it is very unlikely for a color to be skipped because the control panel will
+                    // not be rotating that quickly
+                }
+            }
+        }
+    }
+
+    /**
+     * 
+     * @return String color
+     */
+    public Color getColor() {
+        String colorResult = NetworkTables.Sensors.controlPanelColor.getString("ERROR");
+        if (colorResult == "ERROR") {
+            LogKitten.wtf("Control panel color not in network table. ERROR was returned instead.");
+            return null;
+        } else if (colorResult == null) {
+            LogKitten.wtf("Control panel color not in network table. Null was returned instead.");
+            return null;
+        }
+        switch (colorResult) { // TODO: get correct format for strings
+        case "RED":
+            return Color.RED;
+        case "GREEN":
+            return Color.GREEN;
+        case "YELLOW":
+            return Color.YELLOW;
+        case "BLUE":
+            return Color.BLUE;
+        default:
+            return null;
+        }
+    }
+
+    /**
+     * An enum of the different colors on the control panel.
+     */
+    enum Color {
+        RED, YELLOW, BLUE, GREEN;
+    }
+
+    public int getIndex(Color color) {
+        for (int index = 0; index < colorOrder.length; index++) {
+            if (colorOrder[index] == color) {
+                return index;
+            }
+        }
+        return -1;
```

Comment:
Why are you returning -1? To avoid silently throwing errors, I'd recommend actually having this call fail (with an error) when the index doesn't match any of them